test/e2e-go/restAPI/simulate/simulateRestAPI_test.go

```diff
@@ -0,0 +1,2220 @@
+// source: https://github.com/ahangsu/c2c-testscript/blob/master/c2c_test/max_depth/app.py
+const maxDepthTealApproval = `#pragma version 8
+txn ApplicationID
+int 0
+==
+bnz main_l6
+txn NumAppArgs
+int 1
+==
+bnz main_l3
+err
+main_l3:
+global CurrentApplicationID
+app_params_get AppApprovalProgram
+store 1
+store 0
+global CurrentApplicationID
+app_params_get AppClearStateProgram
+store 3
+store 2
+global CurrentApplicationAddress
+acct_params_get AcctBalance
+store 5
+store 4
+load 1
+assert
+load 3
+assert
+load 5
+assert
+int 2
+txna ApplicationArgs 0
+btoi
+exp
+itob
+log
+txna ApplicationArgs 0
+btoi
+int 0
+>
+bnz main_l5
+main_l4:
+int 1
+return
+main_l5:
+itxn_begin
+  int appl
+  itxn_field TypeEnum
+  int 0
+  itxn_field Fee
+  load 0
+  itxn_field ApprovalProgram
+  load 2
+  itxn_field ClearStateProgram
+itxn_submit
+itxn_begin
+  int pay
+  itxn_field TypeEnum
+  int 0
+  itxn_field Fee
+  load 4
+  int 100000
+  -
+  itxn_field Amount
+  byte "appID"
+  gitxn 0 CreatedApplicationID
+  itob
+  concat
+  sha512_256
+  itxn_field Receiver
+itxn_next
+  int appl
+  itxn_field TypeEnum
+  txna ApplicationArgs 0
+  btoi
+  int 1
+  -
+  itob
+  itxn_field ApplicationArgs
+  itxn CreatedApplicationID
+  itxn_field ApplicationID
+  int 0
+  itxn_field Fee
+  int DeleteApplication
+  itxn_field OnCompletion
+itxn_submit
+b main_l4
+main_l6:
+int 1
+return`
+
+func goValuesToAvmValues(goValues ...interface{}) *[]model.AvmValue {
+	if len(goValues) == 0 {
+		return nil
+	}
+
+	boolToUint64 := func(b bool) uint64 {
+		if b {
+			return 1
+		}
+		return 0
+	}
+
+	modelValues := make([]model.AvmValue, len(goValues))
+	for i, goValue := range goValues {
+		switch converted := goValue.(type) {
+		case []byte:
+			modelValues[i] = model.AvmValue{
+				Type:  uint64(basics.TealBytesType),
+				Bytes: &converted,
+			}
+		case bool:
+			convertedUint := boolToUint64(converted)
+			modelValues[i] = model.AvmValue{
+				Type: uint64(basics.TealUintType),
+				Uint: valToNil(&convertedUint),
+			}
+		case int:
+			convertedUint := uint64(converted)
+			modelValues[i] = model.AvmValue{
+				Type: uint64(basics.TealUintType),
+				Uint: valToNil(&convertedUint),
+			}
+		case basics.AppIndex:
+			convertedUint := uint64(converted)
+			modelValues[i] = model.AvmValue{
+				Type: uint64(basics.TealUintType),
+				Uint: valToNil(&convertedUint),
+			}
+		case uint64:
+			modelValues[i] = model.AvmValue{
+				Type: uint64(basics.TealUintType),
+				Uint: valToNil(&converted),
+			}
+		default:
+			panic("unexpected type inferred from interface{}")
+		}
+	}
+	return &modelValues
+}
+
+func TestMaxDepthAppWithPCandStackTrace(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	a := require.New(fixtures.SynchronizedTest(t))
+	var localFixture fixtures.RestClientFixture
+	localFixture.SetupNoStart(t, filepath.Join("nettemplates", "OneNodeFuture.json"))
+
+	primaryNode, err := localFixture.GetNodeController("Primary")
```

Comment:
note for reviewers, this line (change from global fixture to localFixture) in this and some other test is the fix to these tests (compare with the original `TestMaxDepthAppWithPCandStackTrace` version).
All other changes are splitting a megatest to manageable components.